tests/integration-tests/tests/update/test_update.py

```diff
@@ -529,3 +529,159 @@ def _test_max_vcpus(region, stack_name, vcpus):
+    )
+
+    # Check cfn_scheduler_slots changed by changing compute instance type
+    _check_cfn_scheduler_slots(cluster, scheduler, region, new_compute_instance_type, disable_hyperthreading=True)
+
+
+def _check_ht_update_compute(
+    pcluster_config_reader, clusters_factory, init_config_name, test_datadir, scheduler, region
+):
+    init_config_file = pcluster_config_reader(config_file=init_config_name)
+    cluster = clusters_factory(init_config_file)
+
+    init_config = configparser.ConfigParser()
+    init_config.read(init_config_file)
+
+    # Update cluster with new configuration and change the compute instance type to c5.9xlarge
+    new_compute_instance_type = "c5.9xlarge"
+    _update_config_compute_instance_type(
+        init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, init_config_name
+    )
+
+    _check_cfn_scheduler_slots(cluster, scheduler, region, new_compute_instance_type)
+
+    # Update cluster with new configuration and change the compute instance type to t2.xlarge
+    new_compute_instance_type = "t2.xlarge"
+    _update_config_compute_instance_type(
+        init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, init_config_name
+    )
+
+    _check_cfn_scheduler_slots(cluster, scheduler, region, new_compute_instance_type)
+
+
+def _check_extra_json_update_compute(
+    pcluster_config_reader, clusters_factory, init_config_name, test_datadir, scheduler, region, cfn_scheduler_slots
+):
+    init_config_file = pcluster_config_reader(config_file=init_config_name)
+    cluster = clusters_factory(init_config_file)
+
+    init_config = configparser.ConfigParser()
+    init_config.read(init_config_file)
+
+    # Update cluster with new configuration and change the compute instance type to c5.9xlarge
+    new_compute_instance_type = "c5.9xlarge"
+    _update_config_compute_instance_type(
+        init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, init_config_name
+    )
+
+    _check_cfn_scheduler_slots(
+        cluster, scheduler, region, new_compute_instance_type, cfn_scheduler_slots=cfn_scheduler_slots
+    )
+
+
+def _update_config_compute_instance_type(
+    init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, file_name
+):
+    init_config.set("cluster default", "compute_instance_type", new_compute_instance_type)
+    updated_config_file = pcluster_config_reader(config_file=_save_config_file(init_config, test_datadir, file_name))
+    cluster.config_file = str(updated_config_file)
+    cluster.update()
+
+
+def _save_config_file(config, test_datadir, file_name):
+    try:
+        with open(str(test_datadir / file_name), "w") as configfile:
+            config.write(configfile)
+        return file_name
+    except IOError as err:
+        logging.info("I/O error: {0}".format(err))
```

Comment:
are we just swallowing the error? I'd just let this exception propagate to communicate the failure. probably you don't even need to catch it and log the info cause it'll be logged as an error automatically